### ISSUE TITLE
`ddev composer install` doesn't work with TYPO3 master with mutagen because composer.json depends on project .git

### ISSUE THREAD

**Issue by spoonerWeb** [drud/ddev#3384]
[Diagnostic](https://gist.github.com/spoonerWeb/ce6ae4a6fe47021247f3b3baafd36be4)

**Describe the bug**

Working on TYPO3 core I made a `git pull` and after that a `ddev composer install` to update packages.
I got the error:

```
> ddev composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - typo3/testing-framework is locked to version dev-master and an update of this package was not requested.
    - typo3/testing-framework dev-master requires typo3/cms-backend 12.*.*@dev -> could not be found in any version, there may be a typo in the package name.
  Problem 2
    - typo3/testing-framework dev-master requires typo3/cms-backend 12.*.*@dev -> could not be found in any version, there may be a typo in the package name.
    - typo3/testing-framework is locked to version dev-master and an update of this package was not requested.

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems
```

* Check out TYPO3 master from git: `git clone git@github.com/typo3/typo3` and try to do `ddev composer install`

It works when:
* running `composer install` (with PHP 7.4.26) on host.
* disabling mutagen.

It doesn't work when:
* changing PHP version on ddev, it doesn't work. Tried with PHP 7.4 and 8.0
* doing `ddev mutagen reset` and `ddev start`
* removing `vendor` folder completely

* composer version 2.1.12 (host and ddev both the same version)

**To Reproduce**

No one could reproduce it yet, but:

1. Clone TYPO3 core
2. Run `ddev config`
3. Run `ddev composer install`

**Expected behavior**

Same behaviour like running composer on host (no dependency issues)

**Version and configuration information (please complete the following information):**

- Host computer OS and Version: macOS Monterey 12.0.1
- Docker Desktop version if on macOS or Windows (from "About Docker Desktop"): 4.2.0 (70708)
- ddev version information (use `ddev version`): 1.18.1
- config.yaml contents for the misbehaving project (the part before the comments)
```
name: t3core
type: typo3
docroot: ""
php_version: "7.4"
webserver_type: nginx-fpm
router_http_port: "80"
router_https_port: "443"
xdebug_enabled: false
additional_hostnames: []
additional_fqdns: []
mariadb_version: "10.3"
mysql_version: ""
nfs_mount_enabled: false
mutagen_enabled: true
use_dns_when_possible: true
composer_version: "2"
web_environment: []
```
- Do you have any custom configuration (nginx, php, mysql) in the .ddev folder? No

**Comment by spoonerWeb** [comment]
Done `ddev mutagen status --verbose` after `ddev composer install` and got:
```
Mutagen: ok: 
--------------------------------------------------------------------------------
Name: t3core
Identifier: sync_TZeALYyKHQm9KD3VwqXuI343MvODwTCZPRtSVMiyl0s
Labels: None
Alpha:
	URL: /Users/spooner/Ddev/t3core
	Connection state: Connected
Beta:
	URL: docker://2a1e0acc68ef3951df53dfa41da0ed14731e308ff9862037d39c8a6d0516d024/var/www/html
	Connection state: Connected
Status: Scanning files
--------------------------------------------------------------------------------
```

**Comment by rfay** [comment]
Thanks so much @spoonerWeb - Could you please include how to check out TYPO3 core with the commit you're working against?

**Comment by rfay** [comment]
Also, you said same composer version both in ddev and on host, right? Please show the version.

It's going to be interesting to figure out why this has to do with mutagen, because composer is failing long before anything happens with mutagen, I think.

**Comment by rfay** [comment]
I found out the problem, it's quite simple.

By default, the mutagen integration does not sync the project-level .git directory into the container. This is a safety check to make sure that you can't lose anything related to git. 

However, it's easy to allow .git to sync, you just edit the .ddev/mutagen/mutagen.yml to comment out the exclusion of `.git` and remove the `#ddev-generated` from the top of the file, then `ddev restart`.

I suspect that the TYPO3 composer.json should not be completely dependent on having top-level .git, but don't know. 

Anyway, easy fix.

**Comment by rfay** [comment]
Was that solution adequate for you @spoonerWeb ? If so, let's close this one. But we will get .git bind-mounted into the container where possible.

**Comment by rfay** [comment]
Thanks for reporting this, should be solved with #3507 which just went in.
